prov/opx/src/fi_opx_init.c

```diff
@@ -0,0 +1,481 @@
+		src_addr = calloc(1, sizeof(*src_addr));
+		if (!src_addr) {
+			FI_WARN(fi_opx_global.prov, FI_LOG_FABRIC, "Failed to alloc memory.\n");
+			goto err;
+		}
+		src_addr = mem_dup(&opx_default_addr, sizeof(opx_default_addr));
+	}
+
+	if ((hints != NULL) && (hints->dest_addr != NULL) && (((node == NULL) && (service == NULL)) || (flags & FI_SOURCE))) {
+
+		/*
+		 * man/fi_getinfo.3
+		 *
+		 * dest_addr - destination address
+		 * If specified, indicates the destination address. This field
+		 * will be ignored in hints unless the node and service
+		 * parameters are NULL or FI_SOURCE flag is set. If FI_SOURCE
+		 * is not specified, on output a provider shall return an
+		 * address the corresponds to the indicated node and/or service
+		 * fields, relative to the fabric and domain. Note that any
+		 * returned address is only usable locally.
+		 */
+
+		if ((flags & FI_SOURCE) == 0) {
+			if ((hints->addr_format != FI_FORMAT_UNSPEC) &&
+				(hints->addr_format != FI_ADDR_OPX)) {
+
+				FI_WARN(fi_opx_global.prov, FI_LOG_FABRIC,
+					"invalid addr_format hint (%d)\n", hints->addr_format);
+				errno = FI_EINVAL;
+				return -errno;
+			}
+			dest_addr = mem_dup(hints->dest_addr, hints->dest_addrlen);
+		}
+	}
+
+	fi->dest_addr = dest_addr;
+	fi->src_addr = src_addr;
+
+	/*
+	 * man/fi_fabric.3
+	 *
+	 * On input to fi_getinfo, a user may set this (fi_fabric_attr::fabric)
+	 * to an opened fabric instance to restrict output to the given fabric.
+	 * On output from fi_getinfo, if no fabric was specified, but the user
+	 * has an opened instance of the named fabric, this (fi_fabric_attr::fabric)
+	 * will reference the first opened instance. If no instance has been
+	 * opened, this field will be NULL.
+	 */
+
+	fi->fabric_attr->name = strdup(FI_OPX_FABRIC_NAME);
+	if (!fi->fabric_attr->name) {
+		FI_LOG(fi_opx_global.prov, FI_LOG_DEBUG, FI_LOG_FABRIC,
+				"memory allocation failed");
+		goto err;
+	}
+
+	fi->fabric_attr->prov_version = FI_OPX_PROVIDER_VERSION;
+
+	memcpy(fi->tx_attr, fi_opx_global.default_tx_attr, sizeof(*fi->tx_attr));
+	if (hints && hints->tx_attr) {
+
+		/*
+		 * man/fi_endpoint.3
+		 *
+		 *   fi_tx_attr::caps
+		 *
+		 *   "... If the caps field is 0 on input to fi_getinfo(3), the
+		 *   caps value from the fi_info structure will be used."
+		 */
+		if (hints->tx_attr->caps) {
+			fi->tx_attr->caps = hints->tx_attr->caps;
+		}
+
+		/* adjust parameters down from what requested if required */
+		fi->tx_attr->op_flags = hints->tx_attr->op_flags;
+	} else if (hints && hints->caps) {
+		fi->tx_attr->caps = hints->caps;
+	}
+
+	memcpy(fi->rx_attr, fi_opx_global.default_rx_attr, sizeof(*fi->rx_attr));
+	if (hints && hints->rx_attr) {
+
+		/*
+		 * man/fi_endpoint.3
+		 *
+		 *   fi_rx_attr::caps
+		 *
+		 *   "... If the caps field is 0 on input to fi_getinfo(3), the
+		 *   caps value from the fi_info structure will be used."
+		 */
+		if (hints->rx_attr->caps) {
+			fi->rx_attr->caps = hints->rx_attr->caps;
+		}
```

Comment:
The convention in this repo is not to use braces for a single statement after `if` or `else`, unless when braces are used for a matching `else` or `if` (for example L306-308). This applies to many other locations.